notifications_utils/postal_address.py

```diff
@@ -104,7 +104,7 @@ def has_valid_last_line(self):
     @property
     def has_invalid_characters(self):
         return any(
-            line.startswith(tuple(self.INVALID_CHARACTERS))
+            any([c in line for c in set(self.INVALID_CHARACTERS)])
```

Comment:
It will be a tiny bit faster to use a generator than build up a list in memory, and you can iterate over the string directly – no casting to set needed:
```suggestion
            any(c in line for c in self.INVALID_CHARACTERS)
```

Reply:
Alternatively you could do it with set intersection:
```suggestion
            set(line) & set(self.INVALID_CHARACTERS))
```

Reply:
**Original**
```
python -m timeit -v -s "s = 'abc3def'; q = '12345'" "any([c in s for c in set(q)])"
```
953nsec

***

**Generator** 
```
python -m timeit -v -s "s = 'abc3def'; q = '12345'" "any(c in s for c in q)"
```
680 nsec

***

**Set intersection**
```
python -m timeit -v -s "s = 'abc3def'; q = '12345'" "set(s) & set(q)"
```
827 nsec per loop

Reply:
Actually you can remove the parent loop now, because you don’t need to look at each line individually.

So you can replace the whole method with:
```python
    return any(c in self.normalised_lines for c in self.INVALID_CHARACTERS)
```